src/asynk/async_queue.rs

```diff
@@ -648,67 +583,61 @@ where
 
         let metadata = serde_json::to_value(task)?;
 
-        let task: Task = if self.duplicated_tasks {
-            Self::insert_task_query(&mut transaction, metadata, &task.task_type()).await?
-        } else {
-            Self::insert_task_if_not_exist_query(&mut transaction, metadata, &task.task_type())
+        let task: Task = if !task.uniq() {
+            Self::insert_task_query(&mut transaction, metadata, &task.task_type(), Utc::now())
                 .await?
+        } else {
+            Self::insert_task_if_not_exist_query(
+                &mut transaction,
+                metadata,
+                &task.task_type(),
+                Utc::now(),
+            )
+            .await?
         };
 
         transaction.commit().await?;
 
         Ok(task)
     }
 
-    async fn insert_periodic_task(
-        &mut self,
-        task: &dyn AsyncRunnable,
-        timestamp: DateTime<Utc>,
-        period: i64,
-    ) -> Result<PeriodicTask, AsyncQueueError> {
+    async fn schedule_task(&mut self, task: &dyn AsyncRunnable) -> Result<Task, AsyncQueueError> {
         self.check_if_connection()?;
         let mut connection = self.pool.as_ref().unwrap().get().await?;
         let mut transaction = connection.transaction().await?;
-
         let metadata = serde_json::to_value(task)?;
 
-        let periodic_task =
-            Self::insert_periodic_task_query(&mut transaction, metadata, timestamp, period).await?;
-
-        transaction.commit().await?;
-
-        Ok(periodic_task)
-    }
-
-    async fn schedule_next_task(
-        &mut self,
-        periodic_task: PeriodicTask,
-    ) -> Result<PeriodicTask, AsyncQueueError> {
-        self.check_if_connection()?;
-        let mut connection = self.pool.as_ref().unwrap().get().await?;
-        let mut transaction = connection.transaction().await?;
-
-        let periodic_task = Self::schedule_next_task_query(&mut transaction, periodic_task).await?;
-
-        transaction.commit().await?;
-
-        Ok(periodic_task)
-    }
-
-    async fn fetch_periodic_tasks(
-        &mut self,
-        error_margin: StdDuration,
-    ) -> Result<Option<Vec<PeriodicTask>>, AsyncQueueError> {
-        self.check_if_connection()?;
-        let mut connection = self.pool.as_ref().unwrap().get().await?;
-        let mut transaction = connection.transaction().await?;
-
-        let periodic_task =
-            Self::fetch_periodic_tasks_query(&mut transaction, error_margin).await?;
+        let scheduled_at = match task.cron() {
+            Some(scheduled) => match scheduled {
+                CronPattern(cron_pattern) => {
+                    let schedule = match Schedule::from_str(&cron_pattern) {
```

Comment:
let's implement `From` trait for Schedule so you don't have to match

Reply:
[Comments adressed](https://github.com/ayrat555/fang/pull/71/commits/e32b0fe5b77adb1ff9ec8a9453335b37a00bfccc)